runtime/jcl/common/system.c

```diff
@@ -130,14 +129,22 @@ Java_java_lang_System_getSysPropBeforePropertiesInitialized(JNIEnv *env, jclass
 #if defined(OSX)
 		sysPropValue = "UTF-8";
 #else
-		sysPropValue = getPlatformFileEncoding(env, property, sizeof(property), sysPropID);
+		{
+			char property[128] = {0};
```

Comment:
You can't move this declaration to the inner scope, it's used as the return value from getPlatformFileEncoding() in some cases.

Reply:
Do you refer another `getPlatformFileEncoding()` at L146? There is `char property[128] = {0};` there as well. I felt moving the defines closer to the usage instead of adding an error-prone `#if !defined(OSX) & (JAVA_SPEC_VERSION < 18)` to the original `property` define.

Reply:
You can't declare a stack variable in an inner scope and then use that variable in the outer scope. The inner scope is declaring an array, passing a pointer to the array to getPlatformFileEncoding(), and then getPlatformFileEncoding() can return that same pointer, which gets assigned to sysPropValue. So sysPropValue  can point to the `char property[128]` array declared in the inner scope, and sysPropValue is used in the outer scope.

Reply:
Okay, I missed the `sysPropValue` points to `property`, added a comment before its declaration.
Fixed.